lib/ansible/modules/apt_key.py

```diff
@@ -266,8 +274,7 @@ def main():
             id=dict(type='str'),
             url=dict(type='str'),
             data=dict(type='str'),
-            file=dict(type='path'),
-            key=dict(type='str'),
+            file=dict(type='path', aliases=['filename']),
```

Comment:
Few lines behind there is:
```
filename = module.params['file']
```
And this was in `test/sanity/ignore.txt`:
```
lib/ansible/modules/apt_key.py validate-modules:parameter-type-not-in-doc
lib/ansible/modules/apt_key.py validate-modules:undocumented-parameter
```
To remove these two lines in `ignore.txt` I added this alias (both in docs and in code).

Reply:
`filename` is a Python variable inside the module. It is not visible in any way to the user. The module option is called `file`.

I don't know what exactly the warnings were about, but it was not that a Python variable was not mentioned in argspec.

Reply:
This was added because of this in `ignore.txt`:
`lib/ansible/modules/apt_key.py validate-modules:mutually_exclusive-unknown`
In order to remove this ignored validation I added `filename` as an alias to `file`.
Right now it has not been added to file as an alias as ignore line is present in ignore.txt.
Could you please let me know if it needs to change?

Reply:
The error is that `mutually_exclusive` contains `filename`, which is not an option. It should contain `file` instead. (Adding an alias `filename` would be wrong, since then `mutually_exclusive` would not work correctly.)

This is a potentially breaking change, so it needs a changelog fragment and a porting guide entry.

Reply:
OK, so is it appropriate if this gets fixed in this PR or leave it be this way for now?

Reply:
Both is fine, but if you address it, you definitely need a changelog fragment for that change. (Since it potentially breaks people's roles/playbooks if they implicitly relied on that specifying `file` together with other options works. It's probably better to have this in a separate PR, that cannot be backported to stable-2.10.)

Reply:
Will fix it in another PR then.

Reply:
Could be further discussed in https://github.com/ansible/ansible/pull/70492.